File: internal/models/deploy/change_event.go
package deploy

import "time"

// ChangeEvent 统一变更事件
type ChangeEvent struct {
	ID              uint      `gorm:"primarykey" json:"id"`
	CreatedAt       time.Time `json:"created_at"`
	EventType       string    `gorm:"size:30;not null;index" json:"event_type"` // deploy, nacos_release, sql_ticket, pipeline_run, promotion, release
	EventID         uint      `gorm:"not null" json:"event_id"`
	Title           string    `gorm:"size:300;not null" json:"title"`
	Description     string    `gorm:"type:text" json:"description"`
	ApplicationID   *uint     `gorm:"index" json:"application_id"`
	ApplicationName string    `gorm:"size:100" json:"application_name"`
	Env             string    `gorm:"size:30;index" json:"env"`
	Status          string    `gorm:"size:30" json:"status"`
	RiskLevel       string    `gorm:"size:20" json:"risk_level"`
	Operator        string    `gorm:"size:100" json:"operator"`
	OperatorID      uint      `json:"operator_id"`
	Metadata        string    `gorm:"type:text" json:"metadata"` // JSON 扩展字段
}

func (ChangeEvent) TableName() string { return "change_events" }

File: web/src/services/changeEvent.ts
import request from './api'
import type { ApiResponse, PageResponse } from '../types'

export interface ChangeEvent {
  id: number
  created_at: string
  event_type: string
  event_id: number
  title: string
  description: string
  application_id?: number
  application_name: string
  env: string
  status: string
  risk_level: string
  operator: string
  operator_id: number
  metadata: string
}

export interface ChangeEventFilter {
  event_type?: string
  application_id?: number
  env?: string
  status?: string
  operator?: string
  start_time?: string
  end_time?: string
  page?: number
  page_size?: number
}

export interface EventTypeStat {
  event_type: string
  count: number
}

export const changeEventApi = {
  list(params: ChangeEventFilter) {
    return request.get<any, PageResponse<ChangeEvent>>('/change-events', { params })
  },
  stats() {
    return request.get<any, EventTypeStat[]>('/change-events/stats')
  },
  listByApp(appId: number, limit?: number) {
    return request.get<any, ChangeEvent[]>(`/change-events/by-app/${appId}`, { params: { limit } })
  },
}
